### PR TITLE
Added `example/shapenet.py`

### PR DESCRIPTION
**Pull request by taiya**

### CODE REVIEW

File: examples/shapenet.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+asset_source = kb.AssetSource(source_path)
+ids = list(asset_source.db.loc[asset_source.db['id'].str.startswith('02691156')]['id'])
+asset_id = rng.choice(ids) #< e.g. 02691156_10155655850468db78d106ce0a280f87
+obj = asset_source.create(asset_id=asset_id)
+logging.info(f"selected '{asset_id}'")
+
+# --- make object flat on X/Y and not penetrate floor
+obj.quaternion = kb.Quaternion(axis=[1,0,0], degrees=90)
+# HACK: bounds are not updated after rotation! supposed to be obj.bounds[0][2]
+obj.position = obj.position - (0, 0, obj.bounds[0][1])
```
Comment (Qwlouse) [inline examples/shapenet.py:59]:
```suggestion
obj.position = obj.position - (0, 0, obj.aabbox[0][1])  
```

Comment (Qwlouse) [reply examples/shapenet.py:59]: The bounds are pre-transformation, here you want the aabbox or bbox_3d property instead.

File: examples/shapenet.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# --- WARNING: this path is not yet public
+source_path = "gs://kubric-public/ShapeNetCore.v2"
+
+# --- CLI arguments (and modified defaults)
+parser = kb.ArgumentParser()
+parser.set_defaults(
+  seed=1,
+  frame_start=1,
+  frame_end=30,
+  resolution=(256, 256),
```
Comment (Qwlouse) [inline examples/shapenet.py:16]: this does nothing, instead set width and height

File: kubric/__init__.py [hunk 3]
```diff
@@ -47,15 +47,9 @@
...
-from kubric.core.traits import Vector3D
-from kubric.core.traits import Scale
-from kubric.core.traits import Quaternion
-from kubric.core.traits import RGB
-from kubric.core.traits import RGBA
-from kubric.core.traits import AssetInstance
-
 from kubric.custom_types import AddAssetFunction
 from kubric.custom_types import PathLike
+from kubric.custom_types import Quaternion
```
Comment (Qwlouse) [inline kubric/__init__.py:52]:
```suggestion
from pyquat import Quaternion
```

File: kubric/custom_types.py [hunk 4]
```diff
@@ -25,3 +26,5 @@
 PathLike = Union[str, tfds.core.ReadWritePath]
 
 ArrayLike = Union[Sequence[float], np.ndarray]
+
+Quaternion = pyquat.Quaternion
```
Comment (Qwlouse) [inline kubric/custom_types.py:30]:
```suggestion
```

custom_types is meant more as a place for type-hinting  types

File: kubric/custom_types.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 from typing import Any, Callable, Union, Sequence
 
 import numpy as np
+import pyquaternion as pyquat
```
Comment (Qwlouse) [inline kubric/custom_types.py:19]:
```suggestion
```

**Review by Qwlouse — APPROVED** [review]

File: kubric/__init__.py (hunk 3 above)
Comment (taiya) [reply kubric/__init__.py:52]: The point of this is to make it clear what's the rotational type used within the library?

File: kubric/custom_types.py (hunk 4 above)
Comment (taiya) [reply kubric/custom_types.py:30]: in that case, should we have another "types" module, where all the default types are specified?

**Pull request merged**